assembl/static2/js/app/components/administration/discussion/preferences/index.jsx

```diff
@@ -14,54 +14,101 @@ import SectionTitle from '../../../../components/administration/sectionTitle';
 import { load, postLoadFormat } from './load';
 import { save, createMutationsPromises } from './save';
 import validate from './validate';
+import TextFieldAdapter from '../../../form/textFieldAdapter';
+import { slugAllowedCharacters } from '../../../../constants';
 
-export type DiscussionPreferencesFormProps = {
+type Props = {
   client: ApolloClient,
   locale: string
 };
 
+type State = {
+  slugIsInvalid: boolean
+};
+
 const loading = <Loader />;
 
-const DiscussionPreferencesForm = ({ client, locale }: DiscussionPreferencesFormProps) => (
-  <div className="discussion-admin admin-box admin-content">
-    <SectionTitle title={I18n.t('administration.menu.preferences')} annotation="" />
-    <LoadSaveReinitializeForm
-      load={(fetchPolicy: FetchPolicy) => load(client, fetchPolicy, locale)}
-      loading={loading}
-      postLoadFormat={postLoadFormat}
-      createMutationsPromises={createMutationsPromises(client)}
-      save={save}
-      validate={validate}
-      mutators={{
-        ...arrayMutators
-      }}
-      render={({ handleSubmit, pristine, submitting, values }) => (
-        <div className="admin-content">
-          <AdminForm handleSubmit={handleSubmit} pristine={pristine} submitting={submitting}>
-            <div className="form-container">
-              <div className="title">
-                <Translate value="administration.languageChoice" />
-              </div>
-              <div className="margin-l" />
-              <Field component={CheckboxListFieldAdapter} name="languages" />
-              <div className="separator" />
-              <div className="title">
-                <Translate value="administration.moderation" />
-              </div>
-              <Field
-                component={CheckboxFieldAdapter}
-                name="withModeration"
-                isChecked={values.withModeration}
-                label={I18n.t('administration.activateModeration')}
-                type="checkbox"
-              />
+class DiscussionPreferencesForm extends React.Component<Props, State> {
+  state = {
+    slugIsInvalid: false
+  };
+
+  checkSlugValidity = (slug: ?string) => {
+    if (slug && !slug.match(slugAllowedCharacters)) {
+      this.setState({
```

Comment:
I thought best practice now is to use the async `setState` method

Reply:
Only when you're making use of the previous state or props